src/main/java/ch/njol/skript/effects/EffShear.java

```diff
@@ -18,60 +18,87 @@
  */
 package ch.njol.skript.effects;
 
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.entity.Sheep;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Effect;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
+import io.papermc.paper.entity.Shearable;
+
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Sheep;
+import org.bukkit.entity.Snowman;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
 @Name("Shear")
-@Description("Shears or 'un-shears' a sheep. Please note that no wool is dropped, this only sets the 'sheared' state of the sheep.")
-@Examples({"on rightclick on a sheep holding a sword:",
-		"	shear the clicked sheep"})
-@Since("2.0")
+@Description({
+	"Shears or un-shears a shearable entity with drops by shearing and a 'sheared' sound. Using with 'force' will force this effect despite the entity's 'shear state'.",
+	"\nPlease note that..:",
+	"\n- If your server is not running with Paper 1.19.4 or higher, this effect will only change its 'shear state', and the 'force' effect is unavailable",
+	"\n- Force-shearing or un-shearing on a sheared mushroom cow is not possible"
+})
+@Examples({
+	"on rightclick on a sheep holding a sword:",
+		"\tshear the clicked sheep",
+		"\tchance of 10%",
+		"\tforce shear the clicked sheep"
+})
+@Since("2.0 (sheep & snowmen), INSERT VERSION (all shearable entities)")
+@RequiredPlugins("Paper 1.19.4+ (all shearable entities)")
 public class EffShear extends Effect {
+
+	private static final boolean INTERFACE_METHOD = Skript.classExists("io.papermc.paper.entity.Shearable");
+
 	static {
 		Skript.registerEffect(EffShear.class,
-				"shear %livingentities%",
+				(INTERFACE_METHOD ? "[:force] " : "") + "shear %livingentities%",
 				"un[-]shear %livingentities%");
 	}
 	
 	@SuppressWarnings("null")
```

Comment:
I don't believe this suppress warning is needed

Reply:
It was there in the first place. I'll have it removed regardless.